src/main/scala/com/timeout/healthchecks/package.scala

```diff
@@ -0,0 +1,29 @@
+package com.timeout
+
+import cats.data._
+import cats.syntax.all._
+import cats.std.all._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+package object healthchecks {
+
+  class HealthCheck(val name: String, check: AsyncCheck) {
+
+    val addNameToErrorMessage: String => String = m => s"$name failed: $m"
+
+    def run()(implicit ec: ExecutionContext) = {
+      check().recover {
```

Comment:
If `AsyncCheck` already wraps a `ValidatedNel`, is it correct to recover? It looks like either we allow the check to throw and we recover, or it wraps a validation, which means that the error handling should have been done upfront